home-application/src/main/java/com/softserveinc/ita/homeproject/application/api/ApartmentApiImpl.java

```
@@ -0,0 +1,111 @@
+package com.softserveinc.ita.homeproject.application.api;
+
+import static com.softserveinc.ita.homeproject.application.constants.Permissions.DELETE_OWNERSHIP_PERMISSION;
+import static com.softserveinc.ita.homeproject.application.constants.Permissions.GET_OWNERSHIPS_PERMISSION;
+import static com.softserveinc.ita.homeproject.application.constants.Permissions.GET_OWNERSHIP_PERMISSION;
+import static com.softserveinc.ita.homeproject.application.constants.Permissions.UPDATE_OWNERSHIP_PERMISSION;
+
+import java.math.BigDecimal;
+import javax.validation.Valid;
+import javax.validation.constraints.DecimalMax;
+import javax.validation.constraints.DecimalMin;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+
+import com.softserveinc.ita.homeproject.api.ApartmentsApi;
+import com.softserveinc.ita.homeproject.homeservice.dto.OwnershipDto;
+import com.softserveinc.ita.homeproject.homeservice.service.OwnershipService;
+import com.softserveinc.ita.homeproject.model.CreateApartmentInvitation;
+import com.softserveinc.ita.homeproject.model.ReadOwnership;
+import com.softserveinc.ita.homeproject.model.UpdateApartmentInvitation;
+import com.softserveinc.ita.homeproject.model.UpdateOwnership;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Component;
+
+@Provider
+@Component
+public class ApartmentApiImpl extends CommonApi implements ApartmentsApi {
+
+    @Autowired
+    private OwnershipService ownershipService;
+
+    @Override
+    public Response createInvitation(Long apartmentId, @Valid CreateApartmentInvitation createApartmentInvitation) {
+        return null;
```

Could we throw here UnsupportedOperationException instead of returning null?

---

Sure, changed